vanhouse/src/components/Price.js

```diff
@@ -1,42 +1,97 @@
-import React,{useState}from 'react'
-import {Button} from 'react-bootstrap'
-import UserList from './UserList';
-import '../styles/searchbar.css';
-
-function Price(props){
-    const [state,setState] = useState(0);
-
-    
-    if(state === 0){
-        return(
-            <div className="boxStyle">
-                <h2 className="h2">Price:</h2>
-                <table id="table">
-                    <tr>
-                        <td>from</td>
-                        <td></td>
-                        <td>to</td>
-                    </tr>
-                    <tr>
-                        <td><input type="text"/></td>
-                        <td>-</td>
-                        <td><input type="text"/></td>
-                    </tr>
-                </table>
-                <div className="divStyle">
-                    <Button variant="primary">Submit</Button>
-                    <Button variant="primary" onClick={()=>setState(1)}>Cancel</Button>
-                </div>
-                
-            </div>
-        
-        )
-    }
-    return (
-        <UserList />
-    )
+import PropTypes from "prop-types";
+import React, { useEffect, useState } from "react";
+import { Button } from "react-bootstrap";
+import "../styles/searchbar.css";
+import UserList from "./UserList";
+
+function Price({ setFilterPost, setFilterIdx }) {
+  const [state, setState] = useState(0);
+  const [low, setLow] = useState(0);
+  const [high, setHigh] = useState(0);
+
+
+  function Cancel() {
+
+    fetch("http://localhost:4000/getpost", { method: "GET" })
+      .then((res) => res.json())
+      .then((res) => {
+        console.log(res);
+        setFilterPost(res);
+    setState(1);
```

Comment:
@tyxxxx I think it's a better idea to just store all posts in a variable, and then you just need to call `setFilterPost(allPosts)` when users click cancel button. In this way we don't need to call backend and it's much faster. Same to other cancel operations.